PWGPP/T0/spectrum/AliAnalysisTaskSpectrumT0.cxx

```diff
@@ -0,0 +1,295 @@
+ * documentation strictly for non-commercial purposes is hereby granted   *
+ * without fee, provided that the above copyright notice appears in all   *
+ * copies and that both the copyright notice and this permission notice   *
+ * appear in the supporting documentation. The authors make no claims     *
+ * about the suitability of this software for any purpose. It is          *
+ * provided "as is" without express or implied warranty.                  *
+ **************************************************************************/
+
+// author: Artur Furs, afurs@cern.ch
+// Analysis task for conditional TZERO spectrum analysis
+//
+
+#include "AliAnalysisTaskSpectrumT0.h"
+ClassImp(AliAnalysisTaskSpectrumT0)
+/*******************************************************************************************************************/
+AliAnalysisTaskSpectrumT0::AliAnalysisTaskSpectrumT0(const char *name): AliAnalysisTaskSE(name)
+,mESD(0x0), mOutputList(0x0),mOutputListPerRun(0x0),mEventStruct{}
+,mEventID{},mHistRunStats(0x0)
+{
+  /// Constructor
+
+  /// Define input and output slots here
+  /// Input slot #0 works with a TChain
+  DefineInput(0, TChain::Class());
+  /// Output slot #0 id reserved by the base class for AOD
+  /// Output slot #1 writes into a TH1 container
+  DefineOutput(1, TList::Class());
+}
+/*******************************************************************************************************************/
+AliAnalysisTaskSpectrumT0::~AliAnalysisTaskSpectrumT0()
+{
+// Destructor
+
+  if (mOutputList)	delete mOutputList;
+}
+/*******************************************************************************************************************/
+Bool_t AliAnalysisTaskSpectrumT0::UserNotify()	{
+///
+/// Calls the mother class Notify()
+///
+	return AliAnalysisTaskSE::UserNotify();
+}
+/*******************************************************************************************************************/
+void AliAnalysisTaskSpectrumT0::NotifyRun()
+{
+  #if !(defined(__CINT__) || defined(__MAKECINT__))
+  if(mMapRun2Output.find(fCurrentRunNumber)==mMapRun2Output.end()) {
+    //dummy run
+    auto it = mMapRun2Output.find(0);
+    mRefMapEventID2Output=it->second;
+    mHistRunStats->Fill(0);
+    mRunBin=0;
+  }
+  else {
+    auto it = mMapRun2Output.find(fCurrentRunNumber);
+    mRefMapEventID2Output=it->second;
+    auto itCnt = mMapRun2BinPos.find(fCurrentRunNumber);
+    if(itCnt!=mMapRun2BinPos.end())  {
+      mHistRunStats->Fill(itCnt->second);
+      mRunBin = itCnt->second;
+    }
+  }
+  //InitHists();
+  #endif
+}
+/*******************************************************************************************************************/
+void AliAnalysisTaskSpectrumT0::UserCreateOutputObjects()	{
+  /// Create histograms
+  /// Called once
+  mOutputList = new TList();
+  mOutputList->SetOwner(kTRUE); /// Will delete the histos on cleanup
+  mOutputList->SetName("output");
+
+  #if !(defined(__CINT__) || defined(__MAKECINT__))
```

Comment:
Why do you need this?

Reply:
There are some problems during ROOT5 compilation( mostly with std library). So I decided to hide some parts of the code. I'm planning to use this task only with ROOT6